tests/datatypes/test_tuple.py

```diff
@@ -76,6 +76,14 @@ def test_lt(self):
                     print(p1, "<", p2, ":", p1 < p2)
             """, run_in_function=False)
 
+    def test_immutable(self):
+        self.assertCodeExecution("""
+            x = (1, 2)
+            x[0] = 3
+            x[-1] = 3
+            x[2] = 3
+            """, run_in_function=False)
```

Comment:
This is definitely the right set of things to be testing, and the test will pass, but for the wrong reasons.

The test suite runs the sample code in CPython, and then again in VOC, and verifies that both examples produce the same output. This includes *if the example raises an exception*. So, this code will raise an exception on the `x[0]` assignment... and then *stop running*, because an exception was raised. The `x[-1]` and `x[2]` cases will not be tested.

So - the test needs to be modified to *catch* any anticipated exceptions, print a confirmation message if they *are* raised, and print a *failure* message if they are not. Then, the test will pass/fail depending on whether the confirmation messages were received and printed.